src/org/daisy/dotify/formatter/impl/core/TableOfContentsImpl.java

```diff
@@ -133,21 +257,27 @@ public void endEntry() {
 		Iterator<Block> i = filtered.iterator();
 		while (i.hasNext()) {
 			Block b = i.next();
-			if (refIdForBlock.containsKey(b)) {
+			if (entryIdForBlock.containsKey(b)) {
 				continue;
 			}
-			if (tocBlockForBlock.containsKey(b) // this should always be true
-			    && tocBlocksWithDescendantTocEntry.contains(tocBlockForBlock.get(b))) {
+			if (entryTocBlockForBlock.containsKey(b) // this should always be true
+			    && tocBlocksWithDescendantTocEntry.contains(entryTocBlockForBlock.get(b))) {
 				continue;
 			}
 			i.remove();
 		}
 		return filtered;
-	}
+    }
 
-	private void assertInTocEntry() {
+    private void assertInTocEntry() {
 		if (!inTocEntry) {
-			throw new RuntimeException("Inline content only allowed within toc-entry");
+			throw new RuntimeException("This inline content is only allowed within toc-entry");
```

Comment:
What kind of inline content is allowed within something that is not a tocEntry?

Reply:
I have reverted this change.